src/WebJobs.Script/Description/Proxy/ProxyFunctionInvoker.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.AppService.Proxy.Client.Contract;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    internal class ProxyFunctionInvoker : FunctionInvokerBase
+    {
+        private IProxyClient _proxyClient;
+
+        public ProxyFunctionInvoker(ScriptHost host, FunctionMetadata functionMetadata, IProxyClient proxyClient) : base(host, functionMetadata)
+        {
+            _proxyClient = proxyClient;
+        }
+
+        protected override async Task InvokeCore(object[] parameters, FunctionInvocationContext context)
+        {
+            await _proxyClient.CallAsync(parameters, null, context.Logger);
```

Comment:
It assumes that its just a1 element array which contains the HttpRequestMessage. The response is set in the  request.Properties.

Yes the routing happens once in functions to find the functionMetadata and again in proxies to find the exact route. This is fine, and in a way needed as in proxies we store the proxy as operations (which are verb + route) where as these would be the same function.

---
In reply to: [132558688](https://github.com/Azure/azure-webjobs-sdk-script/pull/1749#discussion_r132558688) [](ancestors = 132558688)